### PR TITLE
feat(specification): submit specification - application layer

### PR DESCRIPTION
**Pull request by kgajowy**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/apps/api/src/modules/specification/application/__tests__/fixtures.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      imports: [CqrsModule],
+      providers: [
+        {
+          provide: SpecificationRepository,
+          useClass: InMemorySpecificationRepo,
+        },
+        SubmitSpecificationHandler,
+      ],
+    }).compile()
+  ).init();
```
Comment (Dyostiq) [inline api/apps/api/src/modules/specification/application/__tests__/fixtures.ts:27]:
```suggestion
  const sandbox = await Test.createTestingModule({
      imports: [CqrsModule],
      providers: [
        {
          provide: SpecificationRepository,
          useClass: InMemorySpecificationRepo,
        },
        SubmitSpecificationHandler,
      ],
    }).compile();
  await sandbox.init();
```

File: api/apps/api/src/modules/specification/application/__tests__/fixtures.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+              operation: SpecificationOperation.Stratification,
+              baseFeatureId: v4(),
+              againstFeatureId: v4(),
+            },
+          ],
+        }),
+      )),
+    ThenItSavesTheSpecification() {
+      expect(specificationId).toBeDefined();
+      expect(repo.getById(specificationId)).toBeDefined();
```
Comment (Dyostiq) [inline api/apps/api/src/modules/specification/application/__tests__/fixtures.ts:55]: I'd check the entire #memory

**Review by Dyostiq — APPROVED** [review]

File: api/apps/api/src/modules/specification/application/__tests__/fixtures.ts (hunk 2 above)
Comment (kgajowy) [reply api/apps/api/src/modules/specification/application/__tests__/fixtures.ts:55]: @Dyostiq you mean if there isn't any other scenarioId or/and the specification shape as well?

Comment (Dyostiq) [reply api/apps/api/src/modules/specification/application/__tests__/fixtures.ts:55]: any other scenarioId I think

Comment (kgajowy) [reply api/apps/api/src/modules/specification/application/__tests__/fixtures.ts:54]: https://github.com/Vizzuality/marxan-cloud/pull/403/commits/fd39d4aa4c60a847ba0489234863375575dd5b97

File: api/apps/api/src/modules/specification/application/__tests__/fixtures.ts (hunk 1 above)
Comment (kgajowy) [reply api/apps/api/src/modules/specification/application/__tests__/fixtures.ts]: https://github.com/Vizzuality/marxan-cloud/pull/403/commits/fd39d4aa4c60a847ba0489234863375575dd5b97

File: api/apps/api/src/modules/specification/commands/specification-input.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  @IsDefined()
+  operation!: SpecificationOperation.Split | SpecificationOperation.Copy;
+
+  @IsUUID()
+  @IsDefined()
+  baseFeatureId!: string;
+}
+
+export class SpecificationFeatureStratification {
+  operation = SpecificationOperation.Stratification;
```
[the remark this answers is not in the document]
Comment (kgajowy) [reply api/apps/api/src/modules/specification/commands/specification-input.ts:40]: Good catch, thanks ! Fixed in https://github.com/Vizzuality/marxan-cloud/pull/403/commits/fd39d4aa4c60a847ba0489234863375575dd5b97

File: api/apps/api/src/modules/specification/commands/specification-input.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  @IsDefined()
+  @IsArray()
+  @ValidateNested()
+  features!: (SpecificationFeature | SpecificationFeatureStratification)[];
+}
+
+export class SpecificationFeature {
+  @IsEnum(SpecificationOperation)
+  @IsDefined()
+  operation!: SpecificationOperation.Split | SpecificationOperation.Copy;
```
[the remark this answers is not in the document]
Comment (kgajowy) [reply api/apps/api/src/modules/specification/commands/specification-input.ts]: Good catch, thanks ! Fixed in https://github.com/Vizzuality/marxan-cloud/pull/403/commits/fd39d4aa4c60a847ba0489234863375575dd5b97

**Review by Dyostiq — APPROVED** [review]

**Pull request merged**